src/test/java/org/elasticsearch/indices/store/IndicesStoreIntegrationTests.java

```diff
@@ -95,6 +95,12 @@ public void indexCleanup() throws Exception {
         assertThat(Files.exists(indexDirectory(node_3, "test")), equalTo(false));
 
         logger.info("--> move shard from node_1 to node_3, and wait for relocation to finish");
+        SlowClusterStateProcessing disruption = null;
+        if (randomBoolean()) {
+            disruption = new SlowClusterStateProcessing(node_3, getRandom(), 0, 0, 1000, 2000);
```

Comment:
I am unsure how. I want to check if it will be deleted even after a small delay, so I would have to put a sleep between startDisruption() and stopDisruption() and this around the reroute below and that does not seem right. Or did you have something else in mind?